src/lightning/data/streaming/downloader.py

```diff
@@ -66,14 +66,15 @@ class LocalDownloader(Downloader):
     def download_file(cls, remote_filepath: str, local_filepath: str) -> None:
         if not os.path.exists(remote_filepath):
             raise FileNotFoundError("The provided remote_path doesn't exist: {remote_path}")
-        shutil.copy(remote_filepath, local_filepath)
+        if remote_filepath != local_filepath:
+            shutil.copy(remote_filepath, local_filepath)
```

Comment:
Is this fixing a bug? We should have a test in test_dataloaders() for these edge cases I think

Reply:
This is more to make sure we can do local testing.